Source/input/PH_GenericInput.cpp

```diff
@@ -2,7 +2,8 @@
 
 void CPH_GenericInput::Update(uint8* ram)
 {
-	for(auto* listener : m_listeners)
+	std::map<unsigned int, std::pair<uint8, uint8>> vibrationMap;
```

Comment:
`std::map` is probably a bit overkill for this. We should use `std::array<std::pair<uint8, uint8>, MAX_PADS>`.

Reply:
updated, I've also added code to choose max value, just incase we implement vibration, in the other interfaces